src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -671,5 +671,33 @@ public static int GetCurrentProcessorId()
         // a speed check will determine refresh rate of the cache and will report if caching is not advisable.
         // we will record that in a readonly static so that it could become a JIT constant and bypass caching entirely.
         private static readonly bool s_isProcessorNumberReallyFast = ProcessorIdCache.ProcessorNumberSpeedCheck();
+
+#if FEATURE_WASM_MANAGED_THREADS
+        [ThreadStatic]
+        public static bool ThrowOnBlockingWaitOnJSInteropThread;
+
+        public static void AssureThreadValidity()
+        {
+            if (ThrowOnBlockingWaitOnJSInteropThread)
+            {
+                throw new PlatformNotSupportedException(SR.WasmThreads_BlockingWaitNotSupportedOnJSInterop);
+            }
+        }
+
+        public static void ForceBlockingWait(Action action)
```

Comment:
This should probably also pass `object? state` so that we could pass `static` lambda, instead of allocating closure.